packages/services/src/view-service/asset-metadata-by-id.ts

```diff
@@ -1,19 +1,25 @@
 import type { Impl } from '.';
 import { servicesCtx } from '../ctx/prax';
-import { getAssetMetadata } from './helpers';
 
-export const assetMetadataById: Impl['assetMetadataById'] = async (req, ctx) => {
-  if (!req.assetId) throw new Error('No asset id passed in request');
-  if (!req.assetId.altBaseDenom && !req.assetId.altBech32m && !req.assetId.inner.length) {
+export const assetMetadataById: Impl['assetMetadataById'] = async ({ assetId }, ctx) => {
+  if (!assetId) throw new Error('No asset id passed in request');
+
+  if (!assetId.altBaseDenom && !assetId.altBech32m && !assetId.inner.length)
     throw new Error(
       'Either `inner`, `altBaseDenom`, or `altBech32m` must be set on the asset ID passed in the `assetMetadataById` request',
     );
-  }
 
   const services = ctx.values.get(servicesCtx);
   const { indexedDb, querier } = await services.getWalletServices();
 
-  const denomMetadata = await getAssetMetadata(req.assetId, indexedDb, querier);
+  const localMetadata = await indexedDb.getAssetsMetadata(assetId);
+  if (localMetadata) return { denomMetadata: localMetadata };
+
+  const remoteMetadata = await querier.shieldedPool.assetMetadataById(assetId);
+  if (remoteMetadata) {
+    void indexedDb.saveAssetsMetadata(remoteMetadata);
+    return { denomMetadata: remoteMetadata };
+  }
 
-  return { denomMetadata };
+  return { denomMetadata: undefined };
```

Comment:
<img width="450" alt="image" src="https://github.com/penumbra-zone/web/assets/25391690/78e7f143-e63a-4355-9c23-b5e3ea699fa5">
Based on the documentation rpc to buf is the correct behavior, we should not throw an error